src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -51,7 +51,7 @@ public User actAs(X x, User entity) throws AuthenticationException {
       throw new AuthenticationException();
     }
```

Comment:
This condition block isn't necessary since subject realUser would always be available.

Reply:
agentAuth is not authenticated, still this check.